### PR TITLE
Refactor exchange collection

### PR DESCRIPTION
**Pull request by ashmrtn**
No real changes to API or internal structure,
but pull out more generic functionality like
getting the path/location info for an
exchange collection into a separate struct
and factor out some soon-to-be common
functions

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #2023

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/exchange/collection.go [hunk 1]
```diff
@@ -36,22 +36,29 @@ const (
...
-	user   string
-	stream chan data.Item
-
-	// added is a list of existing item IDs that were added to a container
-	added map[string]struct{}
-	// removed is a list of item IDs that were deleted from, or moved out, of a container
-	removed map[string]struct{}
-
-	getter itemGetterSerializer
+func NewBaseCollection(
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/exchange/collection.go:39]: This looks generic enough that we could move it into `/internal/data` to share the functionality between other packages.  Thoughts on making that move now?  Not updating the other packages with it.  Just making the move.

File: src/internal/m365/collection/exchange/collection.go [hunk 2]
```diff
@@ -265,33 +337,6 @@ func (col *Collection) streamItems(ctx context.Context, errs *fault.Bus) {
 	wg.Wait()
 }
 
-// finishPopulation is a utility function used to close a Collection's data channel
-// and to send the status update through the channel.
-func (col *Collection) finishPopulation(
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/exchange/collection.go:270]: drat, you got to it before I could!

**Review by ryanfkeepers — APPROVED** [review]
lgtm

File: src/internal/m365/collection/exchange/collection.go (hunk 1 above)
Comment (ashmrtn) [reply src/internal/m365/collection/exchange/collection.go:39]: can we do that in another PR? We can update the other packages at that time. Right now I'd like to just get a minimal set of changes for the Exchange kopia assisted incrementals while touching as few different packages as possible and kinda-sorta setup things for easier code movement in the future (hence the generic-ness of this struct 😉)

Comment (ryanfkeepers) [reply src/internal/m365/collection/exchange/collection.go:39]: fine by me 👍

**Pull request merged**